### PR TITLE
first version of the well testing for physical reason

### PR DESCRIPTION
**Pull request by GitPaean**
the key difficulty is that we do not have reliable explicit information
to do the testing.

In this version, we try to obtain the explicit information by finishing
one converged solving.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/autodiff/MultisegmentWell_impl.hpp [hunk 1]
```diff
@@ -1648,7 +1648,7 @@ namespace Opm
     checkWellOperability(const Simulator& /* ebos_simulator */,
                          const WellState& /* well_state */)
     {
-        const std::string msg = "Support of well operatability checking for mutlisegment wells is not implemented "
+        const std::string msg = "Support of well operability checking for mutlisegment wells is not implemented "
```
Comment (akva2) [inline opm/autodiff/MultisegmentWell_impl.hpp]: may as well fix the typo while at it. multisegment

File: opm/autodiff/MultisegmentWell_impl.hpp [hunk 2]
```diff
@@ -1888,4 +1888,20 @@ namespace Opm
...
+
+
+    template<typename TypeTag>
+    void
+    MultisegmentWell<TypeTag>::
+    wellTestingPhysical(Simulator& simulator, const std::vector<double>& B_avg,
+                        const double simulation_time, const int report_step, const bool terminal_output,
+                        WellState& well_state, WellTestState& welltest_state)
+    {
+        const std::string msg = "Support of well testing for physical limits for mutlisegment wells is not "
```
Comment (akva2) [inline opm/autodiff/MultisegmentWell_impl.hpp]: and here.

Comment (GitPaean) [reply opm/autodiff/MultisegmentWell_impl.hpp]: Thanks for catching them.

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/autodiff/MultisegmentWell.hpp [hunk 3]
```diff
@@ -349,6 +349,10 @@ namespace Opm
         void assembleWellEqWithoutIteration(const Simulator& ebosSimulator,
                                             const double dt,
                                             WellState& well_state);
+
+        virtual void wellTestingPhysical(Simulator& simulator, const std::vector<double>& B_avg,
+                                         const double simulation_time, const int report_step, const bool terminal_output,
+                                         WellState& well_state, WellTestState& welltest_state);
```
Comment (atgeirr) [inline opm/autodiff/MultisegmentWell.hpp]: I get a compiler warning: this function should be marked `override`, same in StandardWell.hpp.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
This PR can be merged, we are only waiting for a final run to complete. @GitPaean if successful, you can merge this.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
We hold this PR for a little longer. It runs through, while some more investigation is needed before merging this.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
This has been tested and found good. We'll merge this now. It's been a long journey!

**Pull request merged**